ddtrace/span.py

```diff
@@ -486,14 +488,21 @@ def pprint(self):
             ("resource", self.resource),
             ("type", self.span_type),
             ("start", self.start),
-            ("end", "" if not self.duration else self.start + self.duration),
-            ("duration", "%fs" % (self.duration or 0)),
+            ("end", None if not self.duration else self.start + self.duration),
+            ("duration", self.duration),
             ("error", self.error),
-            ("tags", ""),
+            ("tags", ",".join("%s:%s" % (str(k), v) for (k, v) in sorted(self.meta.items()))),
+            ("metrics", ",".join("%s:%s" % (str(k), v) for (k, v) in sorted(self.metrics.items()))),
         ]
-
-        lines.extend((" ", "%r:%s" % kv) for kv in sorted(self.meta.items()))
-        return "\n".join("%10s %s" % line for line in lines)
+        return json.dumps(
```

Comment:
what if we did something like:


```python
return json.dumps(
    collections.OrderedDict(
        (k, repr(v) for k, v in data)
    )
)
```


Then we can just keep `("tags": self.meta), ("metrics", self.metrics), ...`

Wonder if:

- Given how it is now if we still can have stringify/encoding issues that cause this function to fail?
- What if a data type is not what we expect? what if there is a numeric value in tags? 

Not going to try and claim I love the look of this, but is this "better" in the sense it might be easier to spot issues?

```python
>>> print(json.dumps(dict(metrics="{!r}".format(dict(a=1, b=dict(a=2), c=3)))))
{"metrics": "{'a': 1, 'b': {'a': 2}, 'c': 3}"}
```


This is probably weird or me being over protective, but, thoughts?

Reply:
Hm, good questions!

Should we represent span in debug logs closer to what is encoded what the attributes themselves (e.g., the resource is `psycopg2.sql.Composed` vs. the encoded text)?

I see more value in presenting the encoded text here. But right now that is handled in different places within the library.

As for the representation of `metrics` and `meta` which are expected to be json arrays, I'll go ahead and print them as such rather than some serialized version.

Reply:
now you bring up a point.... do we want this to be an accurate representation of what the serialized value will look like?

should we use the same code we use to encode the data to send to the tracer?

since we aren't using our `msgpack` encoding here then we might end up in a case where something you have encoded in this method will be different.

Maybe we update this to just do `msgpack` encoding of `self` ?

Reply:
Let's address this in a different PR as this problem is more fundamental than the debug printing here. We might be better off ensuring span attributes are coerced when they are set rather than when we go to write them. We already do this for tags and metrics but should also do this for all attributes.